### ISSUE TITLE
Time serie in wrong order[Help]

### ISSUE THREAD

**Issue by bourbous** [wyguo/ThreeDRNAseq#34]
Hi
I am having trouble with the time serie, my points are never in chronological order. 
I renamed my points (0, 1h, 6h etc.) into T1 T2 T3 T4 as the example. This did not fix the issue. I ordered them in the metadata file, this did not fix it neither. Could you help ?
Thanks a lot
--
Clara

<img width="564" alt="Screen Shot 2022-09-12 at 1 58 20 PM" src="https://user-images.githubusercontent.com/113354171/189649020-9acca609-ec04-4a24-8213-c6c49cce78a7.png">
[metadata.csv](https://github.com/wyguo/ThreeDRNAseq/files/9547794/metadata.csv)

**Comment by wyguo** [comment]
Hi
On the data generation page, how did you select the "Select factor column/columns of interest (multi-select allowed)"?

**Comment by bourbous** [comment]
Thank you for your reply. 
I selected it as attached, is it good ? 

—
Clara Richet-Bourbousse
SECTION GENOMIQUE ENVIRONEMENTALE ET EVOLUTIVE
INSTITUT DE BIOLOGIE DE L’ENS
46 rue d'Ulm • 75 005 PARIS
PH +33 1 44 32 35 23
E bourbous@ ***@***.***>b ***@***.***>io.ens.psl.eu ***@***.***>
www.ibens.ens.fr <http://www.ibens.ens.fr/>

**Comment by wyguo** [comment]
Could you please provide a screenshot of your selection at the step "Step 2: Select factors of experimental design"? From the csv table, I couldn't figure out how did you select the columns in the App for the analysis.

**Comment by bourbous** [comment]
Here it is. 
I have 4 genotypes and 5 time points.
Thanks

—
Clara Richet-Bourbousse
SECTION GENOMIQUE ENVIRONEMENTALE ET EVOLUTIVE
INSTITUT DE BIOLOGIE DE L’ENS
46 rue d'Ulm • 75 005 PARIS
PH +33 1 44 32 35 23
E bourbous@ ***@***.***>b ***@***.***>io.ens.psl.eu ***@***.***>
www.ibens.ens.fr <http://www.ibens.ens.fr/>

**Comment by wyguo** [comment]
Hi, 
Sorry, I didn't see any screenshots. Did you attached any image in your message?

**Comment by bourbous** [comment]
Yes I can see it in my sent message, I try one more time (factor = Time, Genotype, Bio rep = Replicate, quantify = quant.files)


—
Clara Richet-Bourbousse
SECTION GENOMIQUE ENVIRONEMENTALE ET EVOLUTIVE
INSTITUT DE BIOLOGIE DE L’ENS
46 rue d'Ulm • 75 005 PARIS
PH +33 1 44 32 35 23
E bourbous@ ***@***.***>b ***@***.***>io.ens.psl.eu ***@***.***>
www.ibens.ens.fr <http://www.ibens.ens.fr/>

**Comment by wyguo** [comment]
Please try factor=Genotype,Time; Biorep= Replicate. When you perform the time-series analysis, select the genotype as the groups for time-points. It should solve the problem

**Comment by bourbous** [comment]
Thank you, yes now the time points are in order !

I am encountering a second problem though. When I run step 3 time series the screen become grey and the analysis stops. 
On R I can see this :

Gene level 3D trend analysis ... (click "x" to close) (2022-09-12 17:22:12) 
> Limma-voon to estimate mean-vriance trend ... 
Coefficients not estimable: WT.ns2 
> Fit a basic linear model ... 
Coefficients not estimable: WT.ns2 
> Fit the contrast model ... 
[1] "Contrast groups: X3h1.ns1; h1.3.ns1; WT.ns1; X3h1.ns2; h1.3.ns2; WT.ns2”


I have 3 questions : 
- what did I do wrong for the time series analysis ?
- at this stage how to relaunch the analysis without redoing it from the beginning ? (I have seen RData files in the output folder, can we use this on the website ?)
- I do not see the csv with the list of contrasts and DE genes and DAS. Just TPM and counts. Are these generated when doing the final report ? 

Sorry to bother you, we would like to analyze many samples with your great tool so we really want to make it work. 

Best regards,

—
Clara Richet-Bourbousse
SECTION GENOMIQUE ENVIRONEMENTALE ET EVOLUTIVE
INSTITUT DE BIOLOGIE DE L’ENS
46 rue d'Ulm • 75 005 PARIS
PH +33 1 44 32 35 23
E bourbous@ ***@***.***>b ***@***.***>io.ens.psl.eu ***@***.***>
www.ibens.ens.fr <http://www.ibens.ens.fr/>

**Comment by wyguo** [comment]
Hi, 
1. From your metatable information, I think your experimental design does not fit to time-series trend analysis in the 3D RNA-seq App. Time trend analysis is designed for, for example, we have three groups, group1, group2 and group3. In each group, we have the same number of time points T1, T2, T3, T4 and T5. Then can use a curve to fit the time trends and compare how the trends change over time between the groups. But in your case, you have four genotypes and in each genotype, the numbers of time points are different, for example, in 2h1, you only have two time points and in WT you have five time points. I think time-series trend is not the best way for your data analysis. I would recommend you use the normal contrast group settings (The 3D Analysis page on the App) for your data analysis.
2. You need to finish the analysis and go to the final page to save results and generate report. Then all the results and reports will be saved. 
3. If you are using the App on our server, you need to restart the analysis everytime. If you are using a local version by running ThreeDRNAseq R package to start the App, you can upload the intermediate_data.RData and start the analysis where you saved the results.

**Comment by bourbous** [comment]
Dear Wenbin

I am sorry I have a last troubleshooting issue.

I cannot change the working directory, I tried using the setwd function on R before launching the 3D app but it does not work.
The issue is that it creates a folder in MacintoshHD/private/tmp on my Mac, but then it cannot find the pictures to generate the report (I checked and all the required pictures are in the correct folders under the path ../../../private/tmp/ etc.), but somehow the app cannot access them. Here is an example of error message in the report :
## Error in knitr::include_graphics(fig2insert): Cannot find the file(s): "../../../private/tmp/file73ba20b2210/figure/Heatmap DE genes.png"
Is there a way to specify a working directory other than MacintoshHD/private/tmp/ ? Or is this due to another issue ? 

Thank you for your help,

Best regards 

—
Clara Richet-Bourbousse
SECTION GENOMIQUE ENVIRONEMENTALE ET EVOLUTIVE
INSTITUT DE BIOLOGIE DE L’ENS
46 rue d'Ulm • 75 005 PARIS
PH +33 1 44 32 35 23
E bourbous@ ***@***.***>b ***@***.***>io.ens.psl.eu ***@***.***>
www.ibens.ens.fr <http://www.ibens.ens.fr/>

**Comment by wyguo** [comment]
Hi,
When did you install the ThreeDRNAseq R package? If it is an old version, could please uninstall and then reinstall it again? I fixed an issue about the workdir several weeks ago.
